solidity/contracts/system/TBTCSystem.sol

```diff
@@ -566,6 +562,33 @@ contract TBTCSystem is Ownable, ITBTCSystem, DepositLog {
         );
     }
 
+    /// @notice Refreshes the minimum bondable value required from the operator
+    /// to join the sortition pool for tBTC. The minimum bondable value is
+    /// equal to the current minimum lot size collateralized 150% multiplied by
+    /// the current BTC price.
+    /// @dev It is recommended to call this function on tBTC initialization and
+    /// after minimum lot size update.
+    function refreshMinimumBondableValue() public {
+        uint256 bondRequirementSatoshis = getMinimumLotSizeSatoshi().mul(
+            initialCollateralizedPercent
+        ).div(100);
+        uint256 bondRequirementWei = _fetchBitcoinPrice().mul(
+            bondRequirementSatoshis
+        );
+        keepFactorySelection.setMinimumBondableValue(bondRequirementWei);
```

Comment:
Since we're setting this on the sortition pool, shouldn't it be a per-operator value? That is, shouldn't it be 50% rather than 150%?

Reply:
The reasoning for using 150% was to set up the bar on a low level but not on the lowest level: we make sure every operator we add to the pool can create a keep only with its members if needed. Given that we talk about the minimum lot size, it sounds to me like a sane expectation especially that we need to account for ETHBTC price fluctuations. Technically, we can lower it to 50% but I am not sure if we want to. -_-

Reply:
The other danger I'm seeing here is that we've really blurred the lines of responsibility on who decides what about bonds:

- The threshold/member count for keeps is static, but it's managed by `DepositFactory`. But it's ultimately passed through `Deposit` to `TBTCSystem` and then finally to the ECDSA keep.
- The math on total bond amount is done in `DepositFunding.sol`, but relies entirely on information that `TBTCSystem` has at its disposal.
- The math that turns the bond amount and n/m count into a per-node bond is done in `BondedECDSAKeep`.

I would suggest that we:
- Move `n`/`m` values into `TBTCSystem`'s constructor and drop them from `DepositFactory`.
- Stop passing these through `DepositFactory` and `Deposit`.
- Have `requestNewKeep` pass those along to `BondedECDSAKeepFactory`.
- Calculate the initial bond requirement that is passed to `BondedECDSAKeepFactory` entirely in `TBTCSystem`, rather than doing it in `Deposit` and passing that info along.
- Reuse the `n`/`m` info and initial bond requirement calculation function in the implementation of `refreshMinimumBondableValue`.

Reply:
> EDIT: one more bullet. In a perfect world, we would pass the initial required bond + the n/m info to BondedECDSAKeepFactory and let it decide what the implications are on minimum bondable value.
>
> The whole point here is that the minimum bondable value is as much as possible an implementation detail of BondedECDSAKeepFactory, the Keep setup is as much as possible an implementation detail of TBTCSystem, and the deposit and deposit factory are entirely concerned with the deposit's overall collateralization info.

I am no sure if I understand correctly what you say but I think this is ~ what's happening now. `TBTCSystem` is calculating the bond required for the whole keep and `BondedECDSAKeepFactory` splits this bond between members. 

tBTC needs to be aware of the minimum unbonded value because the refresh needs to happen outside of the function opening a keep. In the previous approach we were updating the minimum unbonded value based on the bonding parameter passed to `BondedECDSAKeepFactory` when opening a keep but it doesn't work well.

I think there is something I am missing here... 🤔

Reply:
Taking this discussion offline briefly, will summarize the outcome.

Reply:
We decided to allow operators in a sortition pool who can support only one keep (one member in one keep to be precise). For this reason, we decided for `setMinimumBondValue(minimumBondValue, keepThreshold, keepSize)`, in the `BondedECDSAKeepFactory` that will basically take that and divide `minimumBondValue` by `keepSize` to determine the correct value to set on the sortition pool. This lets the factory be the center of determining how it spreads bonds across operators (which it already is when opening a new keep), while passing along the full info needed for the final setting in the sortition pool.

`keep-ecdsa` change is here: https://github.com/keep-network/keep-ecdsa/pull/510